lighthouse/helpers/plates.py

```diff
@@ -68,18 +70,27 @@ def get_centre_prefix(centre_name: str) -> Optional[str]:
         logger.exception(e)
         raise DataError("Multiple centres with the same name")
 
+def find_samples(query: Dict[str, str]) -> Optional[List[Dict[str, Any]]]:
+    samples = app.data.driver.db.samples
 
-def get_samples(plate_barcode: str) -> Optional[List[Dict[str, Any]]]:
-    logger.info(f"Getting all samples for {plate_barcode}")
+    samples_for_barcode = list(samples.find(query))
 
-    samples = app.data.driver.db.samples
+    logger.info(f"Found {len(samples_for_barcode)} samples for {query['plate_barcode']}")
+
+    return samples_for_barcode
 
-    samples_for_barcode = list(samples.find({"plate_barcode": plate_barcode}))
 
-    logger.info(f"Found {len(samples_for_barcode)} samples for {plate_barcode}")
+def get_samples(plate_barcode: str) -> Optional[List[Dict[str, Any]]]:
+
+    samples_for_barcode = find_samples({"plate_barcode": plate_barcode})
```

Comment:
If we dont use get_samples anymore anywhere in code and we are not expecting to use it in future, maybe is better if we remove it and the test to have less lines of code to maintain. We can always write it back if we need it.